File: PlacementCellBackend/Controllers/FeedBack/feedbackoncompanyController.cs
using Microsoft.AspNetCore.Mvc;
using PlacementCellBackend.DTOs;
using PlacementCellBackend.Services.Feedback.Interfaces;

namespace PlacementCellBackend.Controllers.FeedBack
{
    [ApiController]
    [Route("api/[controller]")]
    public class FeedbackOnCompanyController : ControllerBase
    {
        private readonly IFeedbackOnCompanyService _feedbackService;

        public FeedbackOnCompanyController(IFeedbackOnCompanyService feedbackService)
        {
            _feedbackService = feedbackService;
        }

        [HttpGet]
        public async Task<ActionResult<IEnumerable<AlumniFeedBackOnCompanyDTO>>> GetFeedbackOnCompany()
        {
            var feedbacks = await _feedbackService.GetAllFeedbacksAsync();
            return Ok(feedbacks);
        }

        [HttpGet("{id}")]
        public async Task<ActionResult<AlumniFeedBackOnCompanyDTO>> GetFeedback(string id)
        {
            var feedback = await _feedbackService.GetFeedbackByIdAsync(id);
            if (feedback == null)
                return NotFound();
            return Ok(feedback);
        }

        [HttpPost]
        public async Task<ActionResult<AlumniFeedBackOnCompanyCreateDTO>> PostFeedbackOnCompany(AlumniFeedBackOnCompanyCreateDTO feedback)
        {
            var created = await _feedbackService.CreateFeedbackAsync(feedback);
            return CreatedAtAction(nameof(GetFeedback), new { }, created);
        }

        [HttpPut("{id}")]
        public async Task<IActionResult> PutFeedback(int id, AlumniFeedBackOnCompanyCreateDTO updatedFeedback)
        {


            var success = await _feedbackService.UpdateFeedbackAsync(id, updatedFeedback);
            if (!success)
                return NotFound();

            return NoContent();
        }

        [HttpDelete("{id}")]
        public async Task<IActionResult> DeleteFeedback(int id)
        {
            var success = await _feedbackService.DeleteFeedbackAsync(id);
            if (!success)
                return NotFound();

            return NoContent();
        }
    }
}

File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/controllers/FeedBack/feedbackoncompanyController.java
package com.example.placementicsbackend.controllers.FeedBack;

import com.example.placementicsbackend.dtos.AlumniFeedbackOnCompanyCreateDto;
import com.example.placementicsbackend.dtos.AlumniFeedbackOnCompanyDto;
import com.example.placementicsbackend.services.feedback.interfaces.IFeedbackOnCompanyService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/feedback-on-company")
@RequiredArgsConstructor
public class feedbackoncompanyController {

    private final IFeedbackOnCompanyService feedbackService;

    @GetMapping
    public ResponseEntity<List<AlumniFeedbackOnCompanyDto>> getFeedbackOnCompany() {
        return ResponseEntity.ok(feedbackService.getAllFeedbacks());
    }

    @GetMapping("/{id}")
    public ResponseEntity<AlumniFeedbackOnCompanyDto> getFeedback(@PathVariable String id) {
        AlumniFeedbackOnCompanyDto feedback = feedbackService.getFeedbackById(id);
        if (feedback == null) {
            return ResponseEntity.notFound().build();
        }
        return ResponseEntity.ok(feedback);
    }

    @PostMapping
    public ResponseEntity<AlumniFeedbackOnCompanyDto> postFeedbackOnCompany(
            @RequestBody AlumniFeedbackOnCompanyCreateDto feedback) {
        return ResponseEntity.status(HttpStatus.CREATED).body(feedbackService.createFeedback(feedback));
    }

    @PutMapping("/{id}")
    public ResponseEntity<Void> putFeedback(
            @PathVariable Integer id,
            @RequestBody AlumniFeedbackOnCompanyCreateDto updatedFeedback) {
        if (!feedbackService.updateFeedback(id, updatedFeedback)) {
            return ResponseEntity.notFound().build();
        }
        return ResponseEntity.noContent().build();
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deleteFeedback(@PathVariable Integer id) {
        if (!feedbackService.deleteFeedback(id)) {
            return ResponseEntity.notFound().build();
        }
        return ResponseEntity.noContent().build();
    }
}
